modules/dataquery/php/dataquery.class.inc

```diff
@@ -51,7 +51,25 @@ class Dataquery extends \NDB_Form
         $username    = $user->getUsername();
         $config      = \NDB_Config::singleton();
         $couchConfig = $config->getSetting('CouchDB');
-        $couch       = \NDB_Factory::singleton()->couchDB(
+        $couchConfigValid = isset(
+            $couchConfig['dbName'],
+            $couchConfig['hostname'],
+            $couchConfig['port'],
+            $couchConfig['admin'],
+            $couchConfig['adminpass']
+        );
+        if (!$couchConfigValid) {
+            echo 'Missing CouchDB configuration settings. Cannot load ' .
+                'Data Query Tool.';
```

Comment:
There is no reason to echo inside of the setup of a class, this will just break the returned result from handle().

Reply:
OK, I changed it to just throw a ConfigException. This still breaks the page but it now at least displays 500 instead of a blank page.

The module is odd -- it doesn't use Smarty exactly but also doesn't seem to be properly Reactified/using PSR stuff. I'm not exactly sure how to display an error on the front-end in the right way.

Reply:
Now that you've changed it to throw an exception, lines below the throw in the if statement are unreachable.